### DISCUSSION TITLE
SAVE-CYCLE not working in TINKER9?

### DISCUSSION THREAD

**Discussion by jacekkozuch** [TinkerTools/tinker-gpu#241] [Q&A]
Hi, 

I just noticed that the keyword SAVE-CYCLE is not working in TINKER9. Whatever I do, the arc file is written instead of the .001 and/or .001u etc files (the latter for uinds) that are written in TINKER8. I assume that this feature may have been removed since it may slow down the MD or am I doing something wrong?

Thx!

**Comment by zhi-wang** [comment]
I purposely avoided file I/O as much as I could when I developed the code. Dr. Ponder had put in a lot of effort to make things right in the Fortran code and I just called the "de facto API" in my code.

This is probably an issue in the Fortran code. If not, I need more time to investigate.

**Reply by jacekkozuch** [reply]
Hi Zhi,

hmm, running the same thing on Tinker8 (8.10.5; separate build from Tinker9) 'SAVE-CYCLE' works. So, it may have to do something with the CPU-version included within Tinker9 (what you refer as Fortran code, I suppose) or somewhere the keyword is not properly 'communicated'. 

EDIT:
- SAVE-INDUCED works, meaning the mdinit.f seems to communicate well with the cpp code
- SAVE-CYCLE doesn't work for me
- DCD-ARCHIVE doesn't work for me either
- also I don't know if that's part of a solution: looking through 'rwcrd.cpp', I find an instance of 'else if (output::arcsave)' in line 518, but nothing about 'cyclesave' anywhere where I've looked.

Thanks!

**Reply by jacekkozuch** [reply]
Hi,

I wanted to follow up and ask if you managed to look into the issue that TINKER9 doesn't do SAVE-CYCLE but always writes into an ARC file?

Thx!
Jacek

**Reply by jacekkozuch** [reply]
Btw: I posted this as an issue, so I'll close this discussion
